src/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -15,8 +15,17 @@ uiRoutes
   .defaults(/dashboard/, {
     requireDefaultIndex: true
   })
+  .when('/dashboard', {
+    // For BWC, preserve unsaved dashboard links prior to v5.3
```

Comment:
Might be worth just adding a bit more detail, since it took me a few minutes to understand the implications of the unsaved dashboard link:

```javascript
 // For BWC, preserve unsaved dashboard links (i.e. links with state data and no ID) prior to v5.3
```